docling_core/experimental/serializer/html.py

```diff
@@ -0,0 +1,922 @@
+                if value_id in cell_map:
+                    value_cell = cell_map[value_id]
+                    value_texts.append(html.escape(value_cell.text))
+
+            cell_text = f"<strong>{cell_text}</strong>: {', '.join(value_texts)}"
+
+        # If this cell has children, create a nested list
+        if cell_id in child_links and child_links[cell_id]:
+            children_html = []
+            children_html.append(f"<li>{cell_text}</li>")
+            children_html.append("<ul>")
+
+            for child_id in child_links[cell_id]:
+                children_html.append(
+                    self._render_cell_tree(
+                        cell_id=child_id,
+                        cell_map=cell_map,
+                        child_links=child_links,
+                        value_links=value_links,
+                        level=level + 1,
+                    )
+                )
+
+            children_html.append("</ul>")
+            return "\n".join(children_html)
+
+        elif cell_id in value_links:
+            return f"<li>{cell_text}</li>"
+        else:
+            # Leaf node - just render the cell
+            # return f'<li>{cell_text}</li>'
+            return ""
+
+
+class HTMLKeyValueSerializer(BaseKeyValueSerializer):
+    """HTML-specific key-value item serializer."""
+
+    @override
+    def serialize(
+        self,
+        *,
+        item: KeyValueItem,
+        doc_serializer: "BaseDocSerializer",
+        doc: DoclingDocument,
+        **kwargs,
+    ) -> SerializationResult:
+        """Serializes the passed key-value item to HTML."""
+        if item.self_ref in doc_serializer.get_excluded_refs(**kwargs):
+            return SerializationResult(text="")
+
+        graph_serializer = HTMLGraphDataSerializer()
+
+        # Add key-value if available
+        key_value = graph_serializer.serialize(
+            item=item.graph,
+            doc_serializer=doc_serializer,
+            doc=doc,
+            tag="key-value-region",
+        )
+
+        # Add caption if available
+        caption = doc_serializer.serialize_captions(item=item, **kwargs)
+
+        return SerializationResult(text="\n".join([key_value.text, caption.text]))
+
+
+class HTMLFormSerializer(BaseFormSerializer):
+    """HTML-specific form item serializer."""
+
+    @override
+    def serialize(
+        self,
+        *,
+        item: FormItem,
+        doc_serializer: "BaseDocSerializer",
+        doc: DoclingDocument,
+        **kwargs,
+    ) -> SerializationResult:
+        """Serializes the passed form item to HTML."""
+        if item.self_ref in doc_serializer.get_excluded_refs(**kwargs):
+            return SerializationResult(text="")
+
+        graph_serializer = HTMLGraphDataSerializer()
+
+        # Add key-value if available
+        key_value = graph_serializer.serialize(
+            item=item.graph,
+            doc_serializer=doc_serializer,
+            doc=doc,
+            tag="form-container",
```

Comment:
I don't think we should create our own "tags". in html it is better to use `<div class="form-container"></div>`

Reply:
yes, I think this is just wrong parameter naming, it should be called `class_name`, because we indeed use `<div class="form-container"></div>`